internal/overlord/servstate/handlers.go

```diff
@@ -350,10 +351,44 @@ func (s *serviceData) startInternal() error {
 		// started (previous logs have already been copied).
 		outputIterator = s.logs.HeadIterator(0)
 	}
+
 	serviceName := s.config.Name
 	logWriter := servicelog.NewFormatWriter(s.logs, serviceName)
-	s.cmd.Stdout = logWriter
-	s.cmd.Stderr = logWriter
+
+	plan, err := s.manager.Plan()
+	if err != nil {
+		fmt.Println("failed to retrieve plan configuration")
+	}
+
+	var logDests []*logstate.SyslogWriter
+	if s.config.Logging != nil {
+		for _, name := range s.config.Logging.Destinations {
+			transport, err := s.manager.logMgr.GetTransport(name)
+			if err != nil {
+				return err
+			}
+
+			syslog := logstate.NewSyslogWriter(transport, serviceName)
```

Comment:
So the architecture is one `SyslogTransport` with ring buffer per destination, and then one `SyslogWriter` (a lightweight type) per service per destination?

Reply:
Yes.